src/Models/BaseElement.php

```diff
@@ -714,17 +714,29 @@ public function Link($action = null)
 
     /**
      * @param string|null $action
-     * @return string
+     * @return string|null
      * @throws \Psr\Container\NotFoundExceptionInterface
      * @throws \SilverStripe\ORM\ValidationException
      */
     public function PreviewLink($action = null)
     {
-        $action = $action . '?ElementalPreview=' . mt_rand();
-        $link = $this->Link($action);
-        $this->extend('updatePreviewLink', $link);
+        if ($page = $this->getPage()) {
+            $link = null;
+            if (ClassInfo::hasMethod($page, 'Link')) {
+                $link = $page->Link($action);
+            } else if ($page instanceof CMSPreviewable) {
```

Comment:
Could you, please, change `else if` to `elseif` to get rid of PSR warning.

Reply:
I've changed this logic a bit instead, to make it a little more robust.